tests/core/evaluation/test_marker.py

```diff
@@ -728,3 +728,17 @@ def test_marker_from_path_raises(
 )
 def test_marker_depth(marker: Marker, expected_depth: int):
     assert marker.max_depth() == expected_depth
+
+
+def test_split_sessions(tmp_path):
+    """Tests loading a tracker with multiple sessions."""
+
+    events = [
+        ActionExecuted("action_session_start"),
+        SessionStarted(),
+        UserUttered(intent={"name": "this-intent"}),
+    ]
+    sessions = Marker._split_sessions(events)
+    assert len(sessions) == 2
+    assert len(sessions[0]) == 0
+    assert len(sessions[1]) == len(events)
```

Comment:
I was in the process of testing the solution, but realised that in the original issue I didn't describe it correctly. It looks like `_split_session` creates two sessions, one empty, and one with all the other events including `action exectued: action_start_session`. 
Will look into `_split_session` now to see why this happens, and maybe the fix should be inside this function.

Reply:
To clarify `sessions` should contain only one session of `len(events)`. So the current assertion is just a placeholder while I fix the problem.

Reply:
> Will look into _split_session now to see why this happens

As already explained above:   The `_split_sessions` method partitions the given list of events such that each partition starts with an ActionExecuted event with the name `ACTION_SESSION_START_NAME`.

The motivation for this approach was the way sessions are split by boteval [(slack)](https://rasa-hq.slack.com/archives/C8LEHMTAM/p1635243635080800).  -- So I don't understand why the above example does not start with sessions start event. Can you explain why you constructed the example like this?

Oh, but I think I found a severe bug now: [this](https://rasa.com/docs/rasa/default-actions/#action_session_start) indicates that the slots should be carried over from previous sessions. At the moment, we ignore those slots - while the bot would actually remember those. Will open an issue for that in a bit. @tttthomasssss , FYI

Reply:
>So I don't understand why the above example does not start with sessions start event. Can you explain why you constructed the example like this?

I didn't manually construct the examples, used `rasa shell` and had a normal dialogue with the bot. This resulted in: 
```
ActionExecuted("action_session_start")
SessionStarted()
ActionExecuted("action_listen")
...
```

>As already explained above: The _split_sessions method partitions the given list of events such that each partition starts with an ActionExecuted event with the name ACTION_SESSION_START_NAME.

Sure, but it creates one empty session first. Why is that?

Reply:
Ah, ok, I overlooked that the first action has an "action_session_start" - Please use constants instead of strings (`ACTION_SESSION_START_NAME`). That makes examples easier to parse and ensures consistency / bugs due to typos in the tests.

So the initial issue was just this and not really empty sessions, yes? 

The issue  here is with the first index being interpreted as a session end index. We just have to adapt the parameter in the range there.

Reply:
Yes, I initially thought it was a different problem when I wrote the issue. Sounds like we're on the same page now.

Reply:
Opened issue #10214  for the issues with the slots that I stumbled over above

Reply:
Actually this issue is related to the whole area of "applied event" logic which was basically ignored so far / where we were just told to not use applied events. I'll open another issue just so we have a reminder on that as well.

Reply:
done: #10215